### PR TITLE
Multi tenant compatibility

### PR DESCRIPTION
**Pull request by davidfestal**
### What does this PR do?

This PR introduce the necessary changes to make the `OpenshiftConnector` compatible with a multi-tenancy scenario where workspaces are spead among external Openshift clusters / namespaces with only one main Che server. 

### What issues does this PR fix or reference?

https://github.com/redhat-developer/rh-che/issues/344
https://github.com/redhat-developer/rh-che/issues/345

### CODE REVIEW

**Comment by benoitf** [comment]
ci-build

File: assembly-multiuser/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/MultiUserCheWsMasterModule.java [hunk 1]
```diff
@@ -87,6 +91,6 @@ protected void configure() {
                 + "-addr :4412 "
                 + "-cmd ${SHELL_INTERPRETER} "
                 + "-enable-auth "
-                + "-logs-dir $HOME/che/exec-agent/logs");
+                + "-logs-dir $(eval \"$" + LOGS_DIR_SETTER_VARIABLE + "\"; echo \"$" + LOGS_DIR_VARIABLE + "\")");
```
Comment (garagatyi) [inline assembly-multiuser/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/MultiUserCheWsMasterModule.java]: Can you elaborate on why we need this change?

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java [hunk 2]
```diff
@@ -161,6 +161,19 @@ public DockerConnector(
...
   /**
+   * Gets the API endpoint URL to be used if the `che.workspace.che_server_endpoint` property is
+   * null.
+   *
+   * @return The default API endpoint URL for this connector which is null because docker connector
+   *     uses the default value added in 'che.properties'.
+   * @throws IOException when a problem occurs with docker api calls
+   */
+  @Nullable
+  public String getApiEndpoint() {
```
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java:172]: Can you elaborate on how this relates to other methods of DockerConnector

Comment (davidfestal) [reply plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java:172]: see the comment of commit 2991c3eb64bc39f5a11428ffd06ed2a56c79349e for detailed explanations

File: assembly-multiuser/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/MultiUserCheWsMasterModule.java (hunk 1 above)
Comment (davidfestal) [reply assembly-multiuser/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/MultiUserCheWsMasterModule.java]: see the comment of commit 4bda7b4c9a709483eff25cc0a7628ffaa6cd0c3a for a more detailed explanation

**Comment by davidfestal** [comment]
ci-build

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
ci-build

**Review by skabashnyuk — APPROVED** [review]

**Comment by benoitf** [comment]
will be another PR for che6 branch ?

**Comment by davidfestal** [comment]
I assume yes

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 3]
```diff
@@ -93,22 +93,45 @@ done
...
+CHE_MULTI_USER=${CHE_MULTI_USER:-${DEFAULT_CHE_MULTI_USER}}
+
 if [ "${CHE_MULTI_USER}" == "true" ]; then
   DEFAULT_CHE_KEYCLOAK_DISABLED="false"
-  CHE_DEDICATED_KEYCLOAK=${CHE_DEDICATED_KEYCLOAK:-"true"}
-  DEFAULT_CHE_IMAGE_REPO="docker.io/eclipse/che-server-multiuser"
+  
+  if [ "${CHE_FABRIC8_MULTITENANT}" == "true" ]; then
+    CHE_DEDICATED_KEYCLOAK="false"
+    DEFAULT_CHE_IMAGE_REPO="docker.io/rhchestage/che-server"
```
Comment (l0rd) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: That should be set to: `DEFAULT_CHE_IMAGE_REPO="docker.io/rhchestage/che-server-multiuser"`

Comment (davidfestal) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: Fixed

**Review by garagatyi — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Review by l0rd — APPROVED** [review]
🎉

**Pull request merged**